.devcontainer/devcontainer.json

```diff
@@ -1,9 +1,9 @@
 {
-    "image":  "ghcr.io/microsoft/msquic/linux-build-xcomp",
+    "dockerFile": "Dockerfile",
```

Comment:
Since now you are building the image on the fly, how fast is this?

Reply:
I don't feel such performance difference, because I don't touch the env for 5-10 min after launch...

Reply:
We should make this as fast as possible. You should push the image to ghcr (ideally, a github action to update the image, which I have not done for linux-build-xcomp either) and reference the image path. Then, we should skip the installation step in prepare-machine.ps1.

Reply:
Why need to be as fast as possible?
My understanding of this .devcontainer directory is for codespaces which we do something manually. Not test automation, right?

Reply:
This devcontainer feature is all about dev experience. Starting up codespaces faster is definitely preferable. That's why I used a docker image in the first place.